src/Symfony/Component/Form/Tests/FormErrorIteratorTest.php

```diff
@@ -52,6 +52,43 @@ public function testFindByCodes($code, $violationsCount)
         $this->assertCount($violationsCount, $specificFormErrors);
     }
 
+    /**
+     * @dataProvider findByCodesProvider
+     */
+    public function testFindByCodesWhenAllCausesAreVariousObjectsOrSimpleStrings($code, $violationsCount)
+    {
+        if (!class_exists(ConstraintViolation::class)) {
```

Comment:
we should not require validator component to test the new feature (we aim for compatibility without it)

Reply:
> FYI this test class on master contains one test case

then i suggest to refactor it a bit, and keep a single `testFindByCodes()` where the expectations are built dynamically (if validator exists yes/no)